scheduler/__init__.py

```diff
@@ -1,115 +1,283 @@
-        container_name=container_name,
-        db_uri=db_uri,
-        credential=credential,
-        database_name=database_name
-    )
-    
+    schedules_db_manager = CosmosDBLoader(container_name="schedules")
+    orgs_db_manager = CosmosDBLoader(container_name="organizations")
+    users_db_manager = CosmosDBLoader(container_name="users")
+
+    organizations = check_subscription_statuses(orgs_db_manager.get_organizations())
+    organization_ids = [org["id"] for org in organizations]
+    users_by_org = users_db_manager.get_users_by_organizations(organization_ids)
+
+    logger.info(f"Found organizations: {users_by_org}")
+
     success_count = 0
     try:
         # get all schedules that are active and have a frequency of twice_a_day
-        active_schedules = cosmos_data_loader.get_data(frequency="twice_a_day")
+        active_schedules = schedules_db_manager.get_data(frequency="twice_a_day")
+
         for schedule in active_schedules:
-            logging.info(f"Triggering fetch for schedule {schedule['id']}")
+            logging.info(
+                f"Triggering fetch for schedule {schedule['id']} {schedule['companyId']} {schedule['reportType']}"
+            )
             try:
-                success = trigger_document_fetch(schedule, cosmos_data_loader)
-                if success:
-                    success_count += 1  
+                summarization = trigger_document_fetch(schedule, schedules_db_manager)
+
+                if (summarization["was_summarized_today"]):
+                    success_count += 1
+                    # send email to all users in the organization
+                    for organization, email_list in users_by_org.items():
+                        organization_name = next(
+                            org["name"]
+                            for org in organizations
+                            if org["id"] == organization
+                        )
+
+                        blob_link = summarization["remote_blob_url"]
+                        report_name = f"{summarization['equity_name']} {summarization['financial_type']}"
+
+                        if send_report_email(
+                            blob_link, report_name, organization_name, email_list
+                        ):
+                            logger.info(f"Report {report_name} sent successfully")
+                        else:
+                            logger.error(f"Failed to send email for {report_name}")
+                            logger.error(f"Report with blob link {blob_link} was not sent")
+
             except LastRunUpdateError as e:
                 logging.error(f"Failed to update schedule last run time: {str(e)}")
-                # You might want to implement retry logic here
+                # We might want to implement retry logic here
                 continue
         logging.info(f"Successfully triggered fetch for {success_count} schedules")
     except Exception as e:
```

Comment:
@PedroLabrador The script logs the error but doesn’t re-raise it, which can make debugging harder.

Consider re-raising the exception or implementing a retry mechanism.
```python
except Exception as e:
    logging.error(f"Error in scheduler: {str(e)}")
    raise  # Re-raise for better visibility
  ```

Reply:
No, that log is good enough, It won't make debugging harder, It will only show IF cosmos update did not work, and that "retry mechanism" it would be just for that. Even though retry won't be necessary because function will call itself a few hours later.

Reply:
I see